compiler/compile/OMRCompilation.cpp

```diff
@@ -440,21 +440,29 @@ OMR::Compilation::Compilation(
       _options->setOption(TR_EnableOSROnGuardFailure, false);
       }
 
-   if (_options->getOption(TR_EnableHCR))
+   // Do not default OSR and HCR on if:
+   //  NextGenHCR is disabled, as they are enabled for it
+   //  OSR is explicitly disabled
+   //  FSD is enabled, as HCR cannot be enabled with it
+   //
+   // Defaults are performed this late to account for the disable OSR above, as we do not want to enable HCR in cases where
+   // its not necessary
+   if (!self()->getOption(TR_DisableNextGenHCR) && !self()->getOption(TR_DisableOSR) && !self()->getOption(TR_FullSpeedDebug))
       {
-      if (!self()->getOption(TR_DisableOSR))
-         self()->setOption(TR_EnableOSR); // Make OSR the default for HCR
-      if (_options->getOption(TR_EnableNextGenHCR))
-         {
-         self()->setOption(TR_EnableOSR);
-         _options->setOption(TR_DisableOSR, false);
-         }
+      self()->setOption(TR_EnableOSR); // OSR and HCR must be enabled for NextGenHCR
+      self()->setOption(TR_EnableHCR);
+
+      // TM results in difficult situations when determining where the transition should occur,
+      // so it is currently disabled. The implications of disabling TM should be investigated further.
+      self()->setOption(TR_DisableTM);
```

Comment:
Have we done performance investigations to confirm this is safe on all platforms? Do we plan on addressing this problem in the future?